🐞 fix(setup):

Jesse3692 opened the pull request:
Fixed a specific bug in Windows 10: UnicodeDecodeError

When I was building the application, I encountered the following error:

```python
  Traceback (most recent call last):
    File "d:\github\yifeng-chia\venv\lib\site-packages\pip\_vendor\pep517\in_process\_in_process.py", line 349, in <module>
      main()
    File "d:\github\yifeng-chia\venv\lib\site-packages\pip\_vendor\pep517\in_process\_in_process.py", line 331, in main
      json_out['return_val'] = hook(**hook_input['kwargs'])
    File "d:\github\yifeng-chia\venv\lib\site-packages\pip\_vendor\pep517\in_process\_in_process.py", line 117, in get_requires_for_build_wheel
      return hook(config_settings)
    File "C:\Users\Administrator\AppData\Local\Temp\pip-build-env-zkqkhqel\overlay\Lib\site-packages\setuptools\build_meta.py", line 154, in get_requires_for_build_wheel
      return self._get_build_requires(
    File "C:\Users\Administrator\AppData\Local\Temp\pip-build-env-zkqkhqel\overlay\Lib\site-packages\setuptools\build_meta.py", line 135, in _get_build_requires
      self.run_setup()
    File "C:\Users\Administrator\AppData\Local\Temp\pip-build-env-zkqkhqel\overlay\Lib\site-packages\setuptools\build_meta.py", line 150, in run_setup
      exec(compile(code, __file__, 'exec'), locals())
    File "setup.py", line 112, in <module>
      long_description=open("README.md").read(),
  UnicodeDecodeError: 'gbk' codec can't decode byte 0x9a in position 412: illegal multibyte sequence
```

chang2358 committed:
🐞 fix(setup):

Fixed a specific bug in Windows 10: UnicodeDecodeError

Jesse3692 reviewed (commented):
this pull can resolve UnicodeDecodeError

arvidn commented:
I'm a bit surprised this is an issue in python, but I would think we would want to set the global locale for the process early on, to something that specifies utf-8 as the encoding. I defer to a python expert.

aqk commented:
Thank you for this fix.

Jesse3692 committed:
Update setup.py

Jesse3692 reviewed (commented):
use the same quote style

Jesse3692 committed:
Update setup.py

use 'with' for resource-allocating operations, automatically close file descriptors to release system resources

Jesse3692 reviewed (commented):
use 'with' for resource-allocating operations, automatically close file descriptors to release system resources

Jesse3692 committed:
Update setup.py

Jesse3692 reviewed (commented):
fix a syntax error

Jesse3692 reviewed (commented):
fix a syntax error

hoffmang9 commented:
To fix your lint issue do the following inside the venv.

```
pip install black flake8
black setup.py
flake8 setup.py
```
And check in those changes.

Jesse3692 committed:
Update setup.py

Jesse3692 committed:
Merge branch 'main' into main

Jesse3692 reviewed (commented):
Some code lengths have been changed to solve the length problem.

Jesse3692 committed:
Update setup.py

arvidn commented on setup.py:
```diff
@@ -14,12 +14,15 @@
+    "keyrings.cryptfile==1.3.4",
+    # "keyrings.cryptfile==1.3.8"
+    # Secure storage for keys on Linux (Will be replaced)
+    # See https://github.com/frispete/keyrings.cryptfile/issues/15
```
where these changes deliberate or a rebase-issue?

Jesse3692 replied:
Sorry, my original intention was just to modify the coding error problem, and then found that the reading of the file is also a bit of a problem, so I added a function containing the with to automatically manage the file. But I found that previous submissions had not passed, and one contributor told me that I needed to use black and flake8 to check the file format and code style, so I modified the previous comments.
